src/dmd/console.d

```diff
@@ -121,14 +176,19 @@ struct Console
          */
         void setColor(Color color)
         {
-            enum FOREGROUND_WHITE = FOREGROUND_RED | FOREGROUND_GREEN | FOREGROUND_BLUE;
-            WORD attr = sbi.wAttributes;
-            attr = (attr & ~(FOREGROUND_WHITE | FOREGROUND_INTENSITY)) |
-                   ((color & Color.red)    ? FOREGROUND_RED   : 0) |
-                   ((color & Color.green)  ? FOREGROUND_GREEN : 0) |
-                   ((color & Color.blue)   ? FOREGROUND_BLUE  : 0) |
-                   ((color & Color.bright) ? FOREGROUND_INTENSITY : 0);
-            SetConsoleTextAttribute(handle, attr);
+            if (vtmode)
+                fprintf(_fp, "\033[%d;%dm", color & Color.bright ? 1 : 0, 30 + (color & ~Color.bright));
```

Comment:
Not sure about this duplication of code.

Perhaps define a PosixConsole and WindowsConsole (Windows-only), and decide whether to use WindowsConsole or PosixConsole depending on the console flag?

Reply:
Or maybe something like AnsiConsole and WindowsConsole (legacy)?